### ISSUE TITLE
Allow using atomic types as launch args

### ISSUE THREAD

**Issue by torsteingrindvik** [tracel-ai/cubecl#328]
Currently if using atomics in kernels they need to be passed via some other type e.g. Array<_>.

It's often useful to put in an atomic temporarily for debugging purposes like counting branches in kernels.
That use case would have a better user experience if it's very convenient to add an atomic var as an arg without much hassle.

Therefore it would be nice to easily put in an atomic var directly.

_Some context from discord chat starting here : https://discordapp.com/channels/1038839012602941528/1263876972837339196/1311989480491716699_

**Comment by nathanielsimard** [comment]
I think you can, but the atomic must be in an array or a tensor.

**Comment by torsteingrindvik** [comment]
> I think you can, but the atomic must be in an array or a tensor.

Yes (see the first line of the OP), my hopes were to be able to launch a single atomic without needing wrapper types around it.
